packages/workspace/src/workspace/path_index.ts

```diff
@@ -108,6 +108,20 @@ export const updatePathIndex = (current: PathIndex, unmergedElements: Element[],
 }
 
 export const deserializedPathIndex = (data: string): PathIndex => new PathIndex(JSON.parse(data))
+export const deserializedPathsIndex = (dataEntries: string[]): PathIndex => new PathIndex(
+  _.flatten(dataEntries.map(data => JSON.parse(data)))
```

Comment:
You can use here the builtin `flatMap`
```suggestion
  dataEntries.flatMap(data => JSON.parse(data))
```